app/src/main/java/com/example/surfstop/MainActivity.java

```diff
@@ -28,6 +31,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        scheduleAlarm();
```

Comment:
lets say you open the app for the first time, you schedule your alarm, 15 minutes, itll wake up and trigger again. great, thats the happy case.

but what if before 15 minutes is up, you close the screen and reopen it or you rotate the screen, or anything that triggers an activity recreation. you would end up creating multiple alarm manager intervals and trigger this from firing faster than 15 minutes.